lib/localize.js

```diff
@@ -164,8 +193,21 @@ export const getLocalizeClass = (superclass = class {}) => class LocalizeClass e
 		}
 		if (Object.prototype.hasOwnProperty.call(this, 'getLocalizeResources') || Object.prototype.hasOwnProperty.call(this, 'resources')) {
 			const possibleLanguages = this._generatePossibleLanguages(config);
-			const resourcesPromise = this.getLocalizeResources(possibleLanguages, config);
-			resourcesLoadedPromises.push(resourcesPromise);
+			if (config?.importFunc) {
+				const resourcesPromise = this.getLocalizeResources(possibleLanguages, config);
+				resourcesLoadedPromises.push(resourcesPromise);
+			}
+			if (config?.loadCommon) {
+				resourcesLoadedPromises.push(this.getLocalizeResources(possibleLanguages, {
+					importFunc: async(lang) => {
+						if (commonResources.has(lang)) return commonResources.get(lang);
+						const resources = (await import(`../lang/${lang}.js`)).default;
+						commonResourcesImportCount++;
```

Comment:
This now keeps a local cache of the common resources, which prevents multiple in-memory copies of the same resources for a given language.

Testing that this was actually working was tricky since we can't observe whether a dynamic import happens or not. I went with a simple counter, which is maybe a bit fragile but gets the job done.

Reply:
This is good for now, I think, but the problem is that we make copies of everything by combining them into a single object per-element, which is bad. We should make a separate story to address this before we expand the list of common messages.